src/components/submission-common.js

```diff
@@ -222,42 +229,46 @@ export const ParameterValueTextInput = props => {
     setSource
   } = props
 
-  const updateSourceValueToExpectedType = (primitiveType, value) => {
-    if (isPrimitiveTypeInputValid(primitiveType, value)) {
-      const updatedValue = convertToPrimitiveType(primitiveType, value)
+  const updateSourceValueToExpectedType = value => {
+    const unwrappedType = unwrapOptional(inputType)
+    if (unwrappedType.type === 'primitive' && isPrimitiveTypeInputValid(unwrappedType.primitive_type, value)) {
+      const updatedValue = convertToPrimitiveType(unwrappedType.primitive_type, value)
 
       const newSource = {
         type: source.type,
         parameter_value: updatedValue
       }
       setSource(newSource)
+      return true
     }
   }
 
+  const placeholder = Utils.cond(
+    [unwrapOptional(inputType).type === 'primitive' && unwrapOptional(inputType).primitive_type === 'String', () => '(Empty string)'],
+    [unwrapOptional(inputType).type === 'array', () => Utils.switchCase(unwrapOptional(unwrapOptional(inputType).array_type).primitive_type,
+      ['Int', () => '[1, 2, 3]'],
+      ['Float', () => '[1.5, 2.0, 5]'],
+      ['String', () => '["value1", "value2", "value3"]'],
+      ['Boolean', () => '[true, false, true]']
+    )],
+    () => undefined
```

Comment:
This use of `undefined` to mean "all good" makes me a little nervous for one good and one less-good reason -
- My good reason: If something goes wrong in calculating the message, it's pretty likely that the function will return `undefined` by default and we'll accidentally assume that means "all good"
- My less-good reason: I have a general preference for strong typing and avoiding values like `null`s and `undefined`s as return values from functions, eg by having "active choice" values to mean "no value here".

I wonder what you think about one of these options:
1. Maybe it's a good idea to return a "success" value whenever we validate a value and it looks good? Might make people more inclined to look at what those info message are on slightly questionable inputs if everything else in their table is green?
2. Maybe we can return a "no comment" message type and map that into no change in the input table UI?